book/scheduling-and-threading.md

```diff
@@ -502,16 +470,53 @@ class Tab:
         self.needs_animation_frame = False
 ```
 
-But `render` only does style, layout and paint. We also need raster and draw,
-so add a method to `Browser` and call it from `run_animation_frame`:
+Now, take a look at all the other calls to `render` in your `Tab` and
+`JSContext` methods. Instead of calling `render`, which causes the
+browser to immediately rerun the rendering pipeline, these methods
+should schedule the rendering pipeline to run later. For
+future-proofing, I'm doing to do this in a new `set_needs_render`
+call:
+
+``` {.python expected=False}
+class Tab
+    def set_needs_render(self):
+        self.schedule_animation_frame()
+```
+
+So, for example, the `load` method can call
+`set_needs_render`:[^more-examples]
+
+``` {.python}
+class Tab:
+    def load(self, url, body=None):
+        # ...
+        self.set_needs_render()
+```
+
+As can `innerHTML_set`:
+
+``` {.python}
+class JSContext:
+    def innerHTML_set(self, handle, s):
+        # ...
+        self.tab.set_needs_render()
+```
+
+There are more calls to `render`; you should find and fix all of them.
```

Comment:
This repeats the footnote on line 487.

Reply:
Ah, woops, I meant to move this footnote to the main text.